src/main/java/org/broadinstitute/barclay/help/WDLWorkUnitHandler.java

```diff
@@ -0,0 +1,337 @@
+        final List<Map<String, Object>> allArgsMap = argMap.get("all");
+        // <arg, List<companions>>
+        final Map<String, List<Map<String, Object>>> argCompanionResourceArgMaps = new HashMap<>();
+        allArgsMap.forEach(m -> {
+            final String argName = (String) m.get("name");
+            final Map<String, Object> argPropertiesMap = m;
+            final List<Map<String, Object>> argCompanions = new ArrayList<>();
+            if (companionFiles.containsKey(argName)) {
+                for (final String companion : companionFiles.get(argName)) {
+                    // the companion files retain all properties of the original arg, except for name and synonyms
+                    final Map<String, Object> companionMap = new HashMap<>();
+                    companionMap.putAll(argPropertiesMap);
+                    companionMap.put("name", companion);
+                    companionMap.put("synonyms", "");
+                    companionMap.put("summary", "Companion resource for: " + companionMap.get("summary"));
+                    argCompanions.add(companionMap);
+                }
+                argCompanionResourceArgMaps.put(argName, argCompanions);
+            }
+        });
+
+        // add the property used by the WDL freemarker template for companion resources
+        currentWorkUnit.getRootMap().put(COMPANION_RESOURCES, argCompanionResourceArgMaps);
+    }
+
+    @Override
+    protected String processNamedArgument(
+            final Map<String, Object> argBindings,
+            final NamedArgumentDefinition argDef,
+            final String fieldCommentText) {
+        final String argCategory = super.processNamedArgument(argBindings, argDef, fieldCommentText);
+
+        // replace the java type of the argument with the appropriate wdl type
+        final String wdlType = getWDLTypeForArgument(argDef, (String) argBindings.get("type"));
+        argBindings.put("type", wdlType);
+
+        // now replace the actual argument name with a wdl-friendly name if necessary ("input" and "output"
+        // are reserved words in WDL and can't be used for arg names;  WDL doesn't accept embedded "-" for
+        // variable names, so use a non-kebab name with an underscore
+        final String actualArgName = (String) argBindings.get("name");
+        String wdlName = "--" + WDLTransforms.transformJavaNameToWDLName(actualArgName.substring(2));
+        argBindings.put("name", wdlName);
+
+        // finally, keep track of the outputs
+        final WorkflowResource workflowResource = argDef.getUnderlyingField().getAnnotation(WorkflowResource.class);
+        if (workflowResource != null) {
+            if (workflowResource.output()) {
+                runtimeOutputs.put(wdlName, wdlType);
+            }
+            for (final String companion : workflowResource.companionResources()) {
+                final String companionArgOption = "--" + companion;
```

Comment:
Since it's used more than once, can you pull out the `--` into a prefix variable?

Reply:
I did it locally in this file - it needs to be done more widely, its all over the place in the generator code, but that should done be a separate PR along with the bigger change described in https://github.com/broadinstitute/barclay/issues/163.

Reply:
Sounds good.